### PR TITLE
Moving batch_get_value out of the for loop

### PR DESCRIPTION
**Pull request by IFeelBloated**
Following the issue posted [here](https://github.com/tensorflow/tensorflow/issues/23545), moving the extremely expensive batch_get_value call out of the for loop speeds up weight saving like a million times

### CODE REVIEW

**Comment by IFeelBloated** [comment]
I signed it!

**Comment by fchollet** [comment]
This is trading off computational overhead for memory overhead. I'm not sure this is the right decision.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ymodak: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by fchollet** [comment]
As per our discussion in https://github.com/keras-team/keras/pull/11604, we will not be merging this change. Thank you!
